internal/status/conditions.go

```diff
@@ -51,19 +52,25 @@ func computeGatewayScheduledCondition(gw *gwapiv1b1.Gateway, scheduled bool) met
 }
 
 // computeGatewayReadyCondition computes the Gateway Ready status condition.
-// TODO: Ready condition should surface true when the Envoy Deployment status is ready.
-//       xref: https://github.com/envoyproxy/gateway/issues/345.
-func computeGatewayReadyCondition(gw *gwapiv1b1.Gateway) metav1.Condition {
-	switch len(gw.Status.Addresses) {
-	case 0:
+// Ready condition surfaces true when the Envoy Deployment status is ready.
+func computeGatewayReadyCondition(gw *gwapiv1b1.Gateway, deployment *appsv1.Deployment) metav1.Condition {
+	if len(gw.Status.Addresses) == 0 {
 		return newCondition(string(gwapiv1b1.GatewayConditionReady), metav1.ConditionFalse,
 			string(gwapiv1b1.GatewayReasonAddressNotAssigned),
 			"No addresses have been assigned to the Gateway", time.Now(), gw.Generation)
-	default:
-		return newCondition(string(gwapiv1b1.GatewayConditionReady), metav1.ConditionTrue,
-			string(gwapiv1b1.GatewayReasonReady),
-			"Address assigned to the Gateway", time.Now(), gw.Generation)
 	}
+
+	// If there are any unavailable replicas for the Envoy Deployment, donot
+	// mark the Gateway as ready yet.
+	if deployment.Status.UnavailableReplicas > 0 {
```

Comment:
IMHO the Gateway should report "Ready=true" when `availableReplicas > 0`. Can you include the number of desired replicas (`spec.replicas) and the number of `availableReplicas` in the condition, e.g. `1/2 deployment replicas are available.

Reply:
If 1/n replicas is available, do we want the gateway condition to be `ready=true`?
Based on the comment, I assume that we mark the gateway to be ready in this case.
But then do we still want the message to say the something like:
"Although just 1 replica is available, we're marking it `ready=true`, but  (n-1) replicas are still unavailable"?

Reply:
I have taken care of this in the latest commits, could you PTAL and suggest changes if required. thanks